### ISSUE TITLE
Add `uv run --watch script.py` command to rerun `uv run` on `.py` file changes

### ISSUE THREAD

**Issue by nikitavoloboev** [astral-sh/uv#9652]
Essentially [Bun watch mode](https://bun.sh/docs/runtime/hot) but with `uv`.

**Comment by zanieb** [comment]
Sounds cool!

**Comment by posutsai** [comment]
Hi @nikitavoloboev, 
Do you plan to send a PR? I am also interested in this feature. Not sure if I should start the development.
Thank you.

**Comment by nikitavoloboev** [comment]
I would look into how bun does that and copy that for `uv`. I never had issues there, I assume the only diff is the dep part but I never needed to rerun on dep changes.

**Comment by Gankra** [comment]
@thejchap yes the notify crate (or maybe specifically notify_debouncer_mini which wraps it) is the right tool. The impls in mdbook are in my experience pretty reasonably to base this work off of:

https://github.com/rust-lang/mdBook/blob/master/src/cmd/watch.rs
https://github.com/rust-lang/mdBook/blob/master/src/cmd/watch/native.rs

----

I would prefer if it were cross-platform, I think for the most part all the logic will be portable, so sure get it working on unix first and we'll figure it out from there.

----

I believe for the first draft just taking a script to run is acceptable.

**Comment by thejchap** [comment]
sounds good - and thanks for the mdbook pointers. will get into it

**Comment by thejchap** [comment]
@Gankra got an MVP implementation here: https://github.com/astral-sh/uv/pull/12847

pausing here for some early feedback. i left notes on a couple specific things that came up in the description (around exit codes/clearscreen behavior/debounce behavior), but any general feedback on the direction before i get in further is appreciated

**Comment by thejchap** [comment]
@zanieb @Gankra let me know if i can help/make any more progress on this one

**Comment by AlmogBaku** [comment]
we can use this
https://github.com/watchexec/watchexec/tree/main/crates/lib

**Comment by mgaitan** [comment]
I usually run my app with `uv run my-app` and my source are in a local package (for example under `src/...`), not in a single script file.

Would `--watch` cover that case as well, and not only `scripts.py` style workflows?
If yes, this could remove a lot of custom restart plumbing on our side.

**Comment by zanieb** [comment]
Yeah, it would cover that use case as well.
